### PR TITLE
Update C++ dependencies

### PR DESCRIPTION
**Pull request by snnn**
1. Update google benchmark from 1.8.3 to 1.8.5
2. Update google test from commit in main branch to tag 1.15.0 
3. Update pybind11 from 2.12.0 to 2.13.1
4. Update pytorch cpuinfo to include the support for Arm Neoverse V2, Cortex X4, A720 and A520.
5. Update re2 from 2024-05-01 to 2024-07-02
6. Update cmake to 3.30.1
7. Update Linux docker images
8. Fix a warning in test/perftest/ort_test_session.cc:826:37: error: implicit conversion loses integer precision: 'streamoff' (aka 'long long') to 'const std::streamsize' (aka 'const long') [-Werror,-Wshorten-64-to-32]

### CODE REVIEW

File: cmake/deps.txt [hunk 1]
```diff
@@ -26,9 +26,9 @@ eigen;https://gitlab.com/libeigen/eigen/-/archive/e7248b26a1ed53fa030c5c459f7ea0
 flatbuffers;https://github.com/google/flatbuffers/archive/refs/tags/v23.5.26.zip;59422c3b5e573dd192fead2834d25951f1c1670c
 fp16;https://github.com/Maratyszcza/FP16/archive/0a92994d729ff76a58f692d3028ca1b64b145d91.zip;b985f6985a05a1c03ff1bb71190f66d8f98a1494
 fxdiv;https://github.com/Maratyszcza/FXdiv/archive/63058eff77e11aa15bf531df5dd34395ec3017c8.zip;a5658f4036402dbca7cebee32be57fb8149811e1
-google_benchmark;https://github.com/google/benchmark/archive/refs/tags/v1.8.3.zip;bf9870756ee3f8d2d3b346b24ee3600a41c74d3d
+google_benchmark;https://github.com/google/benchmark/archive/refs/tags/v1.8.5.zip;bf9870756ee3f8d2d3b346b24ee3600a41c74d3d
```
Comment (edgchen1) [inline cmake/deps.txt:29]: the hash didn't change?

**Review by guschmue — APPROVED** [review]
lgtm

File: tools/android_custom_build/Dockerfile [hunk 2]
```diff
@@ -24,7 +24,7 @@ RUN apt-get update && apt-get install --yes --no-install-recommends \
   unzip lsb-release
 
 # cmake
-RUN CMAKE_VERSION=3.27.3 && \
+RUN CMAKE_VERSION=3.30.1 && \
   aria2c -q -d /tmp -o cmake-${CMAKE_VERSION}-linux-x86_64.tar.gz \
   --checksum=sha-256=62e7819fe0867658b6ea765a711686d637dce76cdf6eb0a6b0f1b879e0344fa7 \
```
Comment (edgchen1) [inline tools/android_custom_build/Dockerfile:29]: need to update this checksum too.

Comment (snnn) [reply tools/android_custom_build/Dockerfile]: Thanks!

**Review by jchen351 — APPROVED** [review]

**Comment by snnn** [comment]
Web CI pipeline is failing. This PR will fix it.

**Review by pranavsharma — APPROVED** [review]

**Comment by ranjitshs** [comment]
@snnn 
Latest gtest 1.15.0 is causing test failures in AIX. I am debugging and will raise PR if I see any work-around.

**Comment by snnn** [comment]
Thanks for letting me know

**Pull request merged**